File: output/ChatGPT/codenet/Java/C++/s518927375.cpp
#include <iostream>

using namespace std;

int main() {
    long long k, a, b;
    cin >> k >> a >> b;

    if (b <= a + 2) {
        cout << 1 + k << endl;
        return 0;
    }

    long long x = k - (a - 1);
    long long ans = a + (x / 2) * (b - a) + (x % 2);

    cout << ans << endl;

    return 0;
}

File: output/CodeGen/codenet/Java/Python/s518927375.py
def main():
    k = int(input())
    a = int(input())
    b = int(input())
    if b <= a + 2:
        print(1 + k)
        return
    x = k - (a - 1)
    ans = a + (x // 2) * (b - a) + (x % 2)
    print(ans)

main()